### ISSUE TITLE
aws_fsx_windows_file_system minimum storage capacity does not match console

### ISSUE THREAD

**Issue by pmcevoy** [terraform-providers/terraform-provider-aws#11247]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
 [master ≡ +0 ~1 -0 !]> terraform -v
Terraform v0.12.15
+ provider.aws v2.41.0
```

### Affected Resource(s)

* aws_fsx_windows_file_system 

### Terraform Configuration Files

```hcl
resource "aws_fsx_windows_file_system" "filecluster" {
	storage_capacity    = 32
	throughput_capacity = 128
  active_directory_id = data.terraform_remote_state.directory.outputs.directory_id

  subnet_ids          = [
		data.aws_subnet.service[0].id
	]

  security_group_ids  = [
    data.aws_security_group.vpc.id
  ]
}
```

### Expected Behavior
Support storage capacity as per GUI console:  
> Minimum 32 GiB; Maximum 65536 GiB

### Actual Behavior
```
Error: expected storage_capacity to be in the range (300 - 65536), got 32
```

### References

* https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystem.html#FSx-CreateFileSystem-request-StorageCapacity

> For Windows file systems, valid values are 32 GiB - 65,536 GiB.

**Comment by bflad** [comment]
The lower size is part of an enhancement the service rolled out last month: https://aws.amazon.com/about-aws/whats-new/2019/11/amazon-fsx-for-windows-file-server-reduces-the-minimum-size-for-file-systems-from-300-gbs-to-32-gbs/

**Comment by mgiles-sfdc** [comment]
Param set at line 40 of resource_aws_fsx_file_system.go

`"capacity": {
	Type:         schema.TypeInt,
	Required:     true,
	ForceNew:     true,
	ValidateFunc: validation.IntAtLeast(300),
},
`

**Comment by mrgiles** [comment]
@pmcevoy, @bflad, I wrote a patch for this (#11272), hope it works. It's my first contribution here. I wasn't familiar with the test suite but it seemed to work in the second iteration (I closed the first PR).

**Comment by bflad** [comment]
Validation for the lower `storage_capacity` minimum value has been merged and will release with version 2.43.0 of the Terraform AWS Provider, tomorrow. Thanks to @mrgiles for the implementation. 👍
